OSDependent/Unix/omnisci_path.cpp

```diff
@@ -26,10 +26,30 @@
 #include <boost/filesystem/path.hpp>
 
 #include "Logger/Logger.h"
+#ifdef ENABLE_EMBEDDED_DATABASE
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+#include <dlfcn.h>
+#include <link.h>
+#endif
 
 namespace omnisci {
 
 std::string get_root_abs_path() {
+#ifdef ENABLE_EMBEDDED_DATABASE
+  void* const handle = dlopen(DBEngine_LIBNAME, RTLD_LAZY);
```

Comment:
To my understanding, `ENABLE_EMBEDDED_DATABASE` is effective only when building `libDBEngine.so`.

The omniscidb server itself does not load `libDBEngine.so`, hence the `dlopen` is never called in this case.

Let's consider an application that links against `libDBEngine.so` (such as the Python omniscidbe module). The `get_root_abs_path` function is defined in the `libDBEngine.so` library. When the application calls `get_root_abs_path`, it must mean that the `libDBEngine.so` library has been already loaded and the `dlopen` call here should be a no-op, IIUC.
So, the issue of loading unused library is not valid because calling `get_root_abs_path` is about using the library. Let me know if I misunderstood the issue.

If there is another way to acquire the handle of a loaded library, it could be used as well.

Reply:
The get_root_abs_path function is defined in the source code which is used both for building a libDBEngine.so and a server, utilities (for example, initdb).
Actually, libDBEngine.so is built additionally to the standard omniscidb artifacts in case of ENABLE_EMBEDDED_DATABASE option.

Reply:
Using `dlopen` flag `RTLD_NOLOAD` might resolve the problem.

Reply:
Yes, actually this place determines whether the get_root_abs_path is located in one of the shared objects loaded by the calling application:
if (rc) {
...
} else {
      path_len = readlink("/proc/self/exe", abs_exe_path, sizeof(abs_exe_path));
}

Reply:
> Using `dlopen` flag `RTLD_NOLOAD` might resolve the problem.

Maybe you're right. If the path generated for the libDBEngine is correct for applications that do not use this library (omnisci_server, initdb).

Reply:
Now using RTLD_NOLOAD dlopen: when handler is NULL, fallback to `/proc/self/exe` path to determine the omnisci root path.